fab/sampling_methods/base.py

```diff
@@ -0,0 +1,108 @@
+from typing import Tuple, Optional, Union
+import torch
+
+from fab.types_ import LogProbFunc
+
+
+class Point:
```

Comment:
This is the most central change of this PR, and most other changes relate to this.